src/geometry/manifold/smooth_manifold_with_corners.lean

```diff
@@ -151,6 +151,10 @@ def model_with_corners_self (𝕜 : Type*) [nondiscrete_normed_field 𝕜]
   continuous_to_fun  := continuous_id,
   continuous_inv_fun := continuous_id }
 
+localized "notation `Isf(` 𝕜 `, ` E `)` := model_with_corners_self 𝕜 E" in manifold
```

Comment:
What does `Isf` stand for?

Reply:
`I` is the most common letter used for `model_with_corners` and `sf` stands for `self`. It is like that because @sgouezel suggested `Iself` but I wanted something even shorter.

Reply:
I'm skeptical that any notation which requires a struggle to make work is going to be robust, and to me `model_with_corners_self` is a big improvement over `Isf` (or even `Iself`).

Remember that the reader counts more than the writer when developing new notations. :-) They are not merely to save keystrokes!

Reply:
In this context, this is a statement with full notation
```
instance has_scalar' : has_scalar C∞(𝓘(𝕜, N), N; 𝓘(𝕜), 𝕜) C∞(𝓘(𝕜, N), N; 𝓘(𝕜, V), V)
```
This is a statement with partial notation
```
instance has_scalar' : has_scalar C∞((model_with_corners_self 𝕜 N), N;
  (model_with_corners_self 𝕜 𝕜), 𝕜) C∞((model_with_corners_self 𝕜 N), N;
  (model_with_corners_self 𝕜 V), V)
```
and this is a statement without notation
```
  instance has_scalar' :  has_scalar (times_cont_mdiff_map (model_with_corners_self 𝕜 N)
    (model_with_corners_self 𝕜 𝕜) N 𝕜 ⊤) (times_cont_mdiff_map (model_with_corners_self 𝕜 N)
    (model_with_corners_self 𝕜 V) N V ⊤)
```
I believe the point here is not at all saving keystrokes (I actually never introduced any of my notations for this reason), the point is that statements in differential geometry, which has way more stuff going on formally than many areas of maths, are unreadable without notations. If we take a differential geometry textbook and remove notation, and write everything with long names fulls of underscore, I wonder if anyone could actually understand the statements after a while.

In this precise situation, we will very rarely consider a smooth structure on a vector space that does not come from `model_with_corners_self` and the point of this notation is not to draw all the attention to this, since this is just a technical thing needed for formalization and it is quickly left behind in standard maths. The point is also that the human brain generally recognizes shapes when reading, it does not read every single letter (there's a lot of nice games showing this), so any notation will not represent any effort after having learned it the first time. For the same reason in the case of not using notation, after having learned what `model_with_corners_self` is, when people will read it they will just give a glance to it and they will not read every single letter, which will therefore have no real reason to be there. In addition, I also bet that the first time someone reads `model_with_corners_self` they will right-click anyways it to see what it is since, being a concept which is not explicitly on most geometry books, they will want to verify that it is what they think. So I am not sure it would really save effort even the first time someone sees it.

A similar thing is valid for Iself, I believe that if I wrote Iself Johan would have all the same asked "What does I stand for?" so I am not sure why it is a better idea (again, the first time that someone will see it they will right-click it anyways, no matter what notation is, and the second time, when they learned, the name won't matter, as long as it is something reasonable which is easy to remember). If it were for me, I would use just a symbol related to the letter I such as `𝓘` but if neither this nor `Isf` can be accepted (although again I would not see why) I would at least like anything which is shorter then `model_with_corners_self` such as `Iself`. I've been using the geometry library without notation for more than 2 months and I strongly believe the experience of using it with notation is multiple times better: it is not just quicker to write, but more readable, cleaner and it contains way fewer lines: having to break a statement over multiple lines is distracting. If you are not convinced I can show you a whole file in the two versions to see the difference.

As for parenthesis, sure although I do not see why the notation we used up to know for continuous functions was not a problem: it does exactly what you do not like. If the point is that you accept this for one-letter notation only, then I repropose the notation `𝓘`, which would solve both problems.

Reply:
I vote for `𝓘`

Reply:
Ok cool I fixed it to \MCI. Another idea could be `𝓢` for self, also considering that it is written \MCS that is an acronym of Model with Corners Self.